### PR TITLE
feat(core): handle dpop and client certificate for token exchange

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Implement DPoP and client certificate validation and support for token exchange.

This PR also refactored the implementation in refresh token grant and client credential grant.

There is no additional tests, the code is copied from the original implementation of node-oidc-provider.

## Testing

The existing tests should pass.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/oidc/grants/utils.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      throw new InvalidGrant('mutual TLS client certificate not provided');
+    }
+    // @ts-expect-error -- code from oidc-provider
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    token.setThumbprint('x5t', cert);
+
+    if (originalToken?.['x5t#S256'] && originalToken['x5t#S256'] !== certificateThumbprint(cert)) {
+      throw new InvalidGrant('failed x5t#S256 verification');
+    }
+  }
```
Comment (gao-sun) [inline packages/core/src/oidc/grants/utils.ts:43]: this is not dpop i think

Comment (wangsijie) [reply packages/core/src/oidc/grants/utils.ts]: Updated

File: packages/core/src/oidc/grants/refresh-token.ts [hunk 2]
```diff
@@ -93,8 +90,6 @@ export const buildHandler: (
     },
   } = providerInstance.configuration();
 
-  const dPoP = await dpopValidate(ctx);
```
Comment (gao-sun) [inline packages/core/src/oidc/grants/refresh-token.ts:96]: jc: should we check the dPoP here instead? does the order matter?

**Review by gao-sun — APPROVED** [review]

File: packages/core/src/oidc/grants/refresh-token.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/oidc/grants/refresh-token.ts:96]: The order does not matter, and all the existing tests passed. This change makes the code easier to read.

**Pull request merged**